shell/common/pipeline.h

```diff
@@ -138,6 +138,9 @@ class Pipeline : public fml::RefCountedThreadSafe<Pipeline<R>> {
 
     {
       std::scoped_lock lock(queue_mutex_);
+      if (queue_.empty()) {
```

Comment:
Can you add a comment explaining how it can be empty

Reply:
I was worried about `ProducerCommitFront` popping queue entries. But `depth_` can only be set during the creation of the pipeline.

It was a safety check that should never be triggered. Maybe worth adding it as a `FML_DCHECK` at some point, but i removed it for now.